app/src/main/java/org/sopt/sample/ui/login/LoginViewModel.kt

```diff
@@ -23,8 +24,23 @@ class LoginViewModel @Inject constructor(
     val loginEvent = _loginEvent.asSharedFlow()
     val idText = MutableStateFlow("")
     val pwText = MutableStateFlow("")
-    val isSignUp = combine(idText, pwText) { id, pw ->
-        id.length in 6..10 && pw.length in 8..12
+
+    val isIdValid = idText.map { idText ->
+        idText.matches("^(?=.*[a-zA-Z])(?=.*[0-9]).{6,10}\$".toRegex()) || idText.isEmpty()
+    }.stateIn(
+        started = SharingStarted.WhileSubscribed(5000L),
+        scope = viewModelScope,
+        initialValue = false
+    )
+    val isPwValid = pwText.map { pwText ->
+        pwText.matches("^(?=.*[a-zA-Z])(?=.*[0-9])(?=.*[!@#\$%^&*]).{6,12}\$".toRegex()) || pwText.isEmpty()
+    }.stateIn(
+        started = SharingStarted.WhileSubscribed(5000L),
+        scope = viewModelScope,
+        initialValue = false
+    )
+    val isSignUp = combine(isIdValid, isPwValid, idText, pwText) { isIdValid, isPwValid, id, pw ->
+        isIdValid && isPwValid && id.isNotEmpty() && pw.isNotEmpty()
```

Comment:
와 플로우 쓰면 이런 것도 막 합칠 수 있나보네요
신기합니당
```suggestion
    val isSignIn = combine(isIdValid, isPwValid, idText, pwText) { isIdValid, isPwValid, id, pw ->
        isIdValid && isPwValid && id.isNotEmpty() && pw.isNotEmpty()
```
로그인 뷰니까 `isSignIn` 으로 바꾸는 게 어떤가요

Reply:
좋습니다 ~ 😄